tests/testsuite/run.rs

```diff
@@ -1263,78 +1263,6 @@ fn fail_no_extra_verbose() {
         .run();
 }
 
-#[cargo_test]
-fn run_multiple_packages() {
```

Comment:
> **There is a side effect from this!**  `IncOccurrences` and `StoreValue` would raise a conflict error on multiple occurrences.  All of the new `ArgAction`s instead overwrite previous values.  This PR removes a couple of cases we explicitly test for.

Reply:
Is there any approach to keep the old behaviour? I found that overwriting surprises me somehow.

Reply:
At the moment, there is not a direct way to get the told behavior.  There was a setting `AllArgsOverrideSelf` that [we had talked about making it default](https://github.com/clap-rs/clap/discussions/2627) and instead we made it the only behavior.  I know this matches `argparse`s behavior; unsure what all other parsers.  The idea is that you can always add on to the end of the command to override a previous value, like if someone had defined a cargo alias, they can override parts of the alias rather than get conflict errors.

With 1.64 out, we had changed `--target` to be multiple occurrences which was safe because it was previously an error.  With this new behavior, we'd be switching an override case to a multiple occurrences case which would at minimum require a warning/error/new-behavior transition period.

People can work around this by making a field to take in multiple occurrence (like with `Append`) and then erroring themselves if there is the wrong number of values.  This is  more work for the application and might make the help lie (it might add a `...`, I can't quite remember).

When doing any of this, I hadn't considered the compatibility hazards and wasn't aware of any projects that might care about this behavior.  I could possible add in an inverse of `AllArgsOverrideSelf` if we consider this important for cargo,  Generally my intention for completely removing things is to make the other features easier to discover and to reduce binary size.  At least for binary size, I don't think a toggle for this would add much.

Reply:
I would also prefer to avoid changing the behavior. There are several flags that intentionally only allow a single instance, as multiple instances could be confusing.

For example `-p` in most commands can be used to select multiple packages. However, with this change, some commands will interpret it as ignoring previous flags. For example, `cargo publish -p foo -p bar` would end up being much different than `cargo build -p foo -p bar`.  I would find that confusing, particularly for people who don't very carefully read the documentation.